SimPEG/Maps.py

```diff
@@ -1165,3 +1248,851 @@ def __init__(self, mesh, pts, ptsv=None, order=3, logSigma=True,
         ParametricSplineMap.__init__(self, mesh, pts, ptsv, order, logSigma,
                                      normal)
 
+
+class ParametrizedLayer(IdentityMap):
+    """
+        Parametrized Layer Space
+
+        m = [val_background, val_layer, layer_center, layer_thickness]
+
+
+        .. plot::
+            :include-source:
+
+            from SimPEG import Mesh, Maps, np
+            import matplotlib.pyplot as plt
+
+            fig, ax = plt.subplots(1,1,figsize=(2,3))
+
+            mesh = Mesh.TensorMesh([50,50],x0='CC')
+            mapping = Maps.ParametrizedLayer(mesh)
+            m = np.hstack(np.r_[1., 2., -0.1, 0.2])
+            rho = mapping._transform(m)
+            mesh.plotImage(rho, ax=ax)
+
+        **Required**
+
+        :param Mesh mesh: SimPEG Mesh, 2D or 3D
+
+        **Optional**
+
+        :param float slopeFact: arctan slope factor - divided by the minimum h
+                                spacing to give the slope of the arctan
+                                functions
+        :param float slope: slope of the arctan function
+        :param numpy.ndarray indActive: bool vector with
+
+    """
+
+    slopeFact = 1e2  # will be scaled by the mesh.
+    slope = None
+    indActive = None
+
+    def __init__(self, mesh, **kwargs):
+
+        super(ParametrizedLayer, self).__init__(mesh, **kwargs)
+
+        if self.slope is None:
+            self.slope = self.slopeFact / np.hstack(self.mesh.h).min()
+
+        self.x = [self.mesh.gridCC[:, 0] if self.indActive is None else
+                  self.mesh.gridCC[self.indActive, 0]][0]
+
+        if self.mesh.dim > 1:
+            self.y = [self.mesh.gridCC[:, 1] if self.indActive is None else
+                      self.mesh.gridCC[self.indActive, 1]][0]
+
+        if self.mesh.dim > 2:
+            self.z = [self.mesh.gridCC[:, 2] if self.indActive is None else
+                      self.mesh.gridCC[self.indActive, 2]][0]
+
+    @property
+    def nP(self):
+        return 4
+
+    @property
+    def shape(self):
+        if self.indActive is not None:
+            return (sum(self.indActive), self.nP)
+        return (self.mesh.nC, self.nP)
+
+    def mDict(self, m):
```

Comment:
@rowanc1 : I would appreciate your thoughts on if/how we might use properties for this